commands/logs.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) OpenFaaS Author(s) 2019. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/openfaas/faas-cli/proxy"
+	"github.com/openfaas/faas-cli/schema"
+	"github.com/spf13/cobra"
+)
+
+var (
+	nowFunc = time.Now
+)
+
+func init() {
+
+	initLogCmdFlags(functionLogsCmd)
+
+	faasCmd.AddCommand(functionLogsCmd)
+}
+
+var functionLogsCmd = &cobra.Command{
+	Use:     `logs <NAME> [--tls-no-verify] [--gateway]`,
+	Aliases: []string{"ls"},
+	Short:   "Tail logs from your functions",
+	Long:    "Tail logs from your functions",
+	Example: `faas-cli logs echo
+faas-cli logs echo --follow=false
+faas-cli logs echo --follow=false --since=10m
+faas-cli logs echo --follow=false --since=2010-01-01T00:00:00Z`,
+	Args:    cobra.ExactArgs(1),
+	RunE:    runLogs,
+	PreRunE: noopPreRunCmd,
+}
+
+func noopPreRunCmd(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+// initLogCmdFlags configures the logs command flags, this allows the developer to
+// reset and reinitialize the flags on the command in unit tests
+func initLogCmdFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	cmd.Flags().BoolVar(&tlsInsecure, "tls-no-verify", false, "Disable TLS validation")
+
+	cmd.Flags().String("instance", "", "filter to a specific function instance")
+	cmd.Flags().String("since", "", "include logs since the given timestamp (RFC3339)")
+	cmd.Flags().String("pattern", "", "filter logs that matching the given pattern")
+	cmd.Flags().Bool("invert", false, "invert the pattern match")
+	cmd.Flags().Bool("follow", true, "tail logs")
+	cmd.Flags().Int("limit", 0, "maximum number of log entries to return, unlimited if <=0 ")
```

Comment:
`kubectl logs` uses `tail` as does `docker service logs`.

I think we should use the same to avoid mental mapping with a short `-n`